config-store-api/src/main/java/io/stargate/config/store/api/ConfigStore.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.config.store.api;
+
+import java.util.Map;
+
+public interface ConfigStore {
+
+  /**
+   * It retrieves the generic {@code Map<String, Object>} for a given extension name.
+   *
+   * <p>If the caller will try to look up the config for an extension that does not have any
+   * setting, it will throw the {@link MissingExtensionSettingsException}. This is a {@link
+   * RuntimeException} and it's the caller's responsibility to catch it and handle gracefully or
+   * propagate upper in the call stack.
+   */
+  Map<String, Object> getConfigForExtension(String extensionName)
```

Comment:
We should probably be working with "module" or "component" as those are more encompassing names. In the event that we store config for something like `core` it wouldn't make sense to call `getConfigForExtension()`

Reply:
I've changed it to a `module`.